TriangleApp/app/src/main/java/triangle/triangleapp/helpers/MediaHelper.java

```diff
@@ -0,0 +1,192 @@
+
+        mMediaRecorder.setOnInfoListener(new MediaRecorder.OnInfoListener() {
+            @Override
+            public void onInfo(MediaRecorder mr, int what, int extra) {
+                // Handle the on duration exceeded
+                if (what == MediaRecorder.MEDIA_RECORDER_INFO_MAX_DURATION_REACHED) {
+                    stopStreaming(false);
+                    startStreaming(true);
+
+                    if (webSocket.isConnected()) {
+                        File file = new File(fileName);
+
+                        int size = (int) file.length();
+                        byte bytes[] = new byte[size];
+                        byte tmpBuff[] = new byte[size];
+                        try {
+                            FileInputStream fis = new FileInputStream(file);
+
+                            int read = fis.read(bytes, 0, size);
+                            if (read < size) {
+                                int remain = size - read;
+                                while (remain > 0) {
+                                    read = fis.read(tmpBuff, 0, remain);
+                                    System.arraycopy(tmpBuff, 0, bytes, size - remain, read);
+                                    remain -= read;
+                                }
+                            }
+                        } catch (IOException e) {
+                            Log.e(TAG, "IoExc", e);
+                        }
+                        webSocket.sendStream(bytes);
+                        file.delete();
+                    }
+                }
+            }
+        });
+
+        mMediaRecorder.setOnErrorListener(new MediaRecorder.OnErrorListener() {
+            @Override
+            public void onError(MediaRecorder mediaRecorder, int what, int extra) {
+                Log.d(TAG, "Error in mediarecorder, " + what + ", " + extra);
+            }
+        });
+
+        mMediaRecorder.setMaxDuration(mediaRecorderMaxDuration);
+    }
+
+    private File getOutputMediaFile(int type) {
+        File mediaStorageDir = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS), "MyCameraApp");
+        if (!mediaStorageDir.exists()) {
+            if (!mediaStorageDir.mkdirs()) {
+                Log.d("MyCameraApp", "failed to create directory");
+                return null;
+            }
+        }
+
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        File mediaFile = null;
+        switch (type) {
+            case MEDIA_TYPE_IMAGE:
+                mediaFile = new File(mediaStorageDir.getPath() + File.separator + "IMG_" + timeStamp + ".jpg");
+                break;
+            case MEDIA_TYPE_VIDEO:
+                mediaFile = new File(mediaStorageDir.getPath() + File.separator + "VID_" + timeStamp + ".mp4");
+                break;
+        }
+        return mediaFile;
+    }
+
+    private void startStreaming(boolean firstStart) {
```

Comment:
Comment toegevoegd aan methode: startStreaming.
/**
* Start streaming from camera to server.
* @param firstStart, Check if videorecorder is started for the first time.
*/